### PR TITLE
Kill completed CallActor, CallExecutionActor and WorkflowActors as per DSDEEPB-1052 plus cleanups

### PR DESCRIPTION
**Pull request by geoffjentry**
edit: removing outdated text, but it won't let me delete comment

### CODE REVIEW

File: src/main/scala/cromwell/engine/workflow/SingleWorkflowRunnerActor.scala [hunk 1]
```diff
@@ -48,10 +48,8 @@ case class SingleWorkflowRunnerActor(wdlSource: WdlSource,
   }
 
   def receive = {
-    case CurrentState(_, state: WorkflowState) if state.isTerminal => handleTermination(state)
```
Comment (geoffjentry) [inline src/main/scala/cromwell/engine/workflow/SingleWorkflowRunnerActor.scala]: AFAICT these were unused

**Comment by Horneth** [comment]
So this 
```scala
when(CallDone) {
     case Event(e, _) =>
       log.warning(s"$tag received unexpected event $e while in state $stateName")
       stay()
   }
```
will never be executed now as we kill the actor during the transition to CallDone right ?

**Comment by geoffjentry** [comment]
@Horneth Right. It should be impossible to hit but we have to put something there for the state anyways so I figured might as well keep it there to be safe.

The alternative is to put a NullFunctions there (or whatever it is called), I'm happy to do either

**Comment by Horneth** [comment]
Ok, maybe we can keep the logging then so we know that if it ever gets there something really went wrong. It also seems strange to me that we need to change the way the WorkflowActor is killed to comply with the tests, and not the other way around, but it does not seem to be a problem to have the workflow manager kill it.
Anyway :+1:  for me, over to @mcovarr

File: src/main/scala/cromwell/engine/workflow/WorkflowActor.scala [hunk 2]
```diff
@@ -265,18 +268,19 @@ case class WorkflowActor(workflow: WorkflowDescriptor,
...
-      case r:Seq[Seq[String]] if r.isEmpty => None
+      case r: Seq[Seq[String]] if r.isEmpty => None
       case _ => Some(TerminalUtil.mdTable(rows, header))
     }
   }
 
   def fetchLocallyQualifiedInputs(call: Call): Future[Map[String, WdlValue]] = {
-    /* TODO: This assumes that each Call has a parent that's a Workflow, but with scatter-gather that might not be the case */
-    val parentWorkflow = call.parent.map {_.asInstanceOf[Workflow]} getOrElse {
+    // TODO: This assumes that each Call has a parent that's a Workflow, but with scatter-gather that might not be the case
```
Comment (mcovarr) [inline src/main/scala/cromwell/engine/workflow/WorkflowActor.scala]: caveat mergor

File: src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala [hunk 3]
```diff
@@ -82,8 +80,10 @@ class WorkflowManagerActor(dataAccess: DataAccess, backend: Backend) extends Act
     case CallOutputs(workflowId, callName) => callOutputs(workflowId, callName) pipeTo sender
     case CallStdoutStderr(workflowId, callName) => callStdoutStderr(workflowId, callName) pipeTo sender
     case WorkflowStdoutStderr(workflowId) => workflowStdoutStderr(workflowId) pipeTo sender
-    case CurrentState(actor, state: WorkflowState) => updateWorkflowState(actor, state)
-    case Transition(actor, oldState, newState: WorkflowState) => updateWorkflowState(actor, newState)
+    case Transition(actor, oldState, newState: WorkflowState) =>
+      updateWorkflowState(actor, newState)
+      // This is hokey but WorkflowActor can't shut itself down due to tests relying on it lasting past completion
```
Comment (mcovarr) [inline src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala]: yuk, which tests and can we not fix those?

File: src/main/scala/cromwell/engine/workflow/WorkflowActor.scala (hunk 2 above)
Comment (geoffjentry) [inline src/main/scala/cromwell/engine/workflow/WorkflowActor.scala]: Note that this branch only changes the comment syntax not the actual TODO.

Comment (mcovarr) [inline src/main/scala/cromwell/engine/workflow/WorkflowActor.scala]: Yup, but still... :smile:

File: src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala (hunk 3 above)
Comment (geoffjentry) [inline src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala]: IIRC it was everything handling WorkflowActor. And as a hakker would say they're all integration tests. The issue is that they set up a workflow actor, run something and then ask it for its outputs

I'm pretty sure that craptastic only-for-testing code path used to be used for other things and now no one gets rid of it

Comment (geoffjentry) [inline src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala]: CromwellTestkitSepc.runWdl
2x in SimpleWorkflowActorSpec

**Comment by geoffjentry** [comment]
Closing this PR. The actual work was about 5 minutes but it turns out that our messed up testing causes doing it right to take a fair amount of effort. I'll leave the ticket for a soon-to-be sprint and someone can pick it up then.

**Pull request closed without merging**